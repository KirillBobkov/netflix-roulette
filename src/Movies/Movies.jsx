import React from 'react';
import { movies } from '../moviesData';
import Movie from '../Movie';
import './Movies.scss'

export default class Movies extends React.PureComponent {
    constructor(props) {
        super(props)
        this.moviesList = movies.map((movie, index) =>
        <Movie key={movie.id} index={index}  info={movie} />
      );
    }

    render() {
        return (
            <div className='movies'>
                <ul className='movies__container'>
                   {this.moviesList}
                </ul>  
            </div>
        );
    }
}
