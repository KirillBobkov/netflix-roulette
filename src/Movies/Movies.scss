.movies {


    &__container {
        display: flex;
        flex-wrap: wrap;
        width: 1060px;
        margin: 0 auto;
        padding: 50px;
    }
}