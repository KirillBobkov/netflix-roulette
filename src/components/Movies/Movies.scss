.movies {


    &__error {
        margin: 0 auto;
        font-size: 30px;
        line-height: 30px;
    }

    &__container {
        display: flex;
        flex-wrap: wrap;
        width: 970px;
        margin: 0 auto;
        padding: 55px 50px 19px 50px;
    }
}