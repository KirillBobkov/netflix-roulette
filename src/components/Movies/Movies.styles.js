import styled from "styled-components";

export const MoviesError = styled.li`
    margin: 0 auto;
    font-size: 30px;
    line-height: 30px;
`;

export const MoviesWrapper = styled.ul`
    display: flex;
    flex-wrap: wrap;
    width: 970px;
    margin: 0 auto;
    padding: 55px 50px 19px 50px;
`;
