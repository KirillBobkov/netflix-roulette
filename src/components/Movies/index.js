export { default as MoviesList } from './MoviesContainer';