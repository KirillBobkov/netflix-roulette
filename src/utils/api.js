import axios from 'axios';

const api = axios.create({
    baseURL: "https://reactjs-cdp.herokuapp.com/",
    responseType: "json"
});

export const getMovies = (params = { 
    sortBy: "release_date", 
    sortOrder: "asc", 
    searchBy: "title" 
}) => {
    return api.get('/movies', {
        params
    });
};

export const getImage = url => {
    return new Promise((resolve, reject) =>{
        var img = new Image();
        img.onload = function(){
            resolve(url);
        };
        img.onerror = function(){
            reject(url);
        };
        img.src = url;
    });
};




