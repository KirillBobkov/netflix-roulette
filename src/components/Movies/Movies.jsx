import React from 'react';
import { Movie } from '../Movie';
import './Movies.scss';
import PropTypes from 'prop-types';

//Create a node list from movies array
export const MoviesItems = ({ list }) => {
    return list.map((movie) => <Movie key={movie.id} movie={movie} />);
};

//Render movies or return fallback message
export const MoviesList = ({ list }) => {
    const hasMovies = Boolean(list && Array.isArray(list) && list.length);

    return (
      <div className='movies'>
        <ul className='movies__container'>
          {hasMovies
              ? <MoviesItems list={list} />
              : <li className='movies__error'>No movies found!</li>}
        </ul>
      </div>
    );
};

MoviesList.defaultProps = {
  movies: null
};

MoviesList.propTypes = {
  list: PropTypes.array
};

