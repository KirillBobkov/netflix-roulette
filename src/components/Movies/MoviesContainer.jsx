import { connect } from 'react-redux';
import { fetchMovies } from '../../store/actions/movieActions';
import { MoviesList } from './Movies';

const mapStateToProps = (state) => ({ list: state.list });

const mapDispatchToProps = dispatch => ({
  fetchMovies: () => dispatch(fetchMovies())
});

export default connect(mapStateToProps, mapDispatchToProps)(MoviesList);