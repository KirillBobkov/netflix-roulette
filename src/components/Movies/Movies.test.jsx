import React from 'react';
import { MoviesItems, MoviesList, MoviesDataWrapper, MoviesListContainer } from './Movies';
import { mount } from 'enzyme';
import { getMovies } from '../../utils';
import { Provider } from 'react-redux';
import { movies } from '../../utils/testData';
import configureStore from 'redux-mock-store';

describe('Movies component', () => {
    const initialState = movies;
    const mockStore = configureStore();
    const mountWithStore = component => mount(<Provider store={store}> {component} </Provider>);
    let store, 
        wrapperMoviesListContainer, 
        wrapperMoviesDataWrapper, 
        wrapperMoviesItems;
    
    beforeAll(()=>{
      store = mockStore(initialState);
      wrapperMoviesListContainer = mountWithStore(<MoviesListContainer />);
      wrapperMoviesDataWrapper = mountWithStore(<MoviesDataWrapper render={MoviesList} />);
      wrapperMoviesItems =  mountWithStore(<MoviesItems list={movies.list} />);
    });
    
    it('should be render MoviesDataWrapper correctly', () => {
        expect(wrapperMoviesDataWrapper).toMatchSnapshot();
    }); 

    it('should be render MoviesListContainer correctly', () => {
        expect(wrapperMoviesListContainer).toMatchSnapshot();
    }); 

    it('should be render movies correctly', () => {
        expect(wrapperMoviesItems).toMatchSnapshot();
    }); 
});   

describe('getMovie fetch service', () => {
    it('should check that method getMovies() return promise', () => {
        expect(movies.list.length).toBeGreaterThan(1);
    }); 
}); 



