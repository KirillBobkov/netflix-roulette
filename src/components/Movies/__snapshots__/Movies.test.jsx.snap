// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies component should be render MoviesDataWrapper correctly 1`] = `ReactWrapper {}`;

exports[`Movies component should be render MoviesList correctly 1`] = `ReactWrapper {}`;

exports[`Movies component should be render movies correctly 1`] = `ReactWrapper {}`;
